src/main/java/com/nhantran/pages/BookTicketPage.java

```diff
@@ -1,41 +1,80 @@
+            By bookTicketSelectBox = By.xpath(String.format(selectBox, comboBoxName.getValue()));
+            SeleniumActions.scrollToElement(bookTicketSelectBox);
+            SeleniumActions.selectByText(bookTicketSelectBox, value);
+        }
     }
 
-    public void selectDepartDate(String date) {
-        this.selectBookTicketInfo("Date", date);
+    private void selectDepartDate(String date) {
+        if (date != null)
+            this.selectBookTicketInfo(BookTicketComboBoxes.DEPART_DATE, date);
     }
 
-    public void selectDepartStation(String departStation) {
-        this.selectBookTicketInfo("DepartStation", departStation);
+    private void selectDepartStation(RailwayStations departStation) {
+        if (departStation != null)
+            this.selectBookTicketInfo(BookTicketComboBoxes.DEPART_STATION, departStation.getValue());
     }
 
-    public void selectArrivalStation(String arrivalStation) {
-        this.selectBookTicketInfo("ArriveStation", arrivalStation);
+    private void selectArrivalStation(RailwayStations arrivalStation) {
+        if (arrivalStation != null)
+            this.selectBookTicketInfo(BookTicketComboBoxes.ARRIVE_STATION, arrivalStation.getValue());
     }
 
-    public void selectSeatType(String seatType) {
-        this.selectBookTicketInfo("SeatType", seatType);
+    private void selectSeatType(SeatTypes seatType) {
+        if (seatType != null)
+            this.selectBookTicketInfo(BookTicketComboBoxes.SEAT_TYPE, seatType.getValue());
     }
 
-    public void selectAmount(Integer amount) {
-        this.selectBookTicketInfo("TicketAmount", String.valueOf(amount));
+    private void selectAmount(Integer amount) {
+        if (amount > 0)
+            this.selectBookTicketInfo(BookTicketComboBoxes.AMOUNT, String.valueOf(amount));
     }
 
-    public void clickBookTicketButton() {
+    private void clickBookTicketButton() {
+        SeleniumActions.scrollToElement(By.xpath(bookTicketButton));
         SeleniumActions.clickElement(By.xpath(bookTicketButton));
     }
 
+    public void bookTicket(Tickets ticket) {
+        selectDepartDate(ticket.getDepartDate());
+        selectDepartStation(ticket.getDepartStation());
+        selectArrivalStation(ticket.getArrivalStation());
+        selectSeatType(ticket.getSeatType());
+        selectAmount(ticket.getTicketAmount());
+        clickBookTicketButton();
+    }
+
+    private static final EnumSet<BookTicketComboBoxes> ALLOWED_STATIONS =
+            EnumSet.of(BookTicketComboBoxes.DEPART_STATION, BookTicketComboBoxes.ARRIVE_STATION);
 
+    public String getStation(BookTicketComboBoxes comboBoxName) {
+        if (!ALLOWED_STATIONS.contains(comboBoxName)) {
+            throw new IllegalArgumentException("Invalid combobox: " + comboBoxName);
+        }
+        return SeleniumActions.getSelectedOption(By.xpath(String.format(selectBox, comboBoxName.getValue())));
+    }
+
+    public String calculateNextDate(int numberOfNextDays) {
```

Comment:
it does not belong to this page. it's better if put it under utils (e/x: DateUtils)